### ISSUE TITLE
Detect unnecessary safe navigation

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#13835]
Annoyed by polluted code and people not understanding what safe navigation operator is and using it everywhere. 
If it is known that a variable can't be `nil`, suggest removing useless safe navigation calls.

```ruby
# bad
x = var.foo # <--- here, if var is nil, an error will be raised in any case
var&.bar    # <--- unnecessary safe navigation

# good
x = var.foo
var.bar

# bad
if var
  var&.foo
end

# good
if var
  var.foo
end
```

**Comment by vlad-pisanov** [comment]
I've proposed something like this too https://github.com/rubocop/rubocop/issues/13142

**Comment by fatkodima** [comment]
Valid.

**Comment by fatkodima** [comment]
Valid.

**Comment by fatkodima** [comment]
@bbatsov @vlad-pisanov I implemented a cop for this locally. It detected 221 offense in my app, and only 1 is a false positive, so this is indeed valuable.

I was about to open a PR, but noticed that there is already a `RedundantSafeNavigation` cop. But the one I am about to create has a different approach and not sure if it is good to integrate it into the existing one. What do you suggest? If a new cop, then how to better name it?

**Comment by vlad-pisanov** [comment]
`RedundantSafeNavigation` would be the most logical place for it imo. The name is just too good.

As a standalone cop, maybe it could be named something like `SafeNavigationOnNonNil`, `SafeNavigationOnKnownValue`? 🤷‍♂
